components/fastly_backends/main.tf

```diff
@@ -14,11 +14,39 @@ variable "papertrail_destination" {
 }
 
 locals {
-  headers = {
+  request_headers = {
+    # Provide geolocation information to web servers:
     "X-Fastly-Country-Code" = "client.geo.country_code",
     "X-Fastly-Region-Code"  = "client.geo.region",
     "X-Fastly-Postal-Code"  = "client.geo.postal_code",
   }
+
+  response_headers = {
+    # Expose geolocation information in debug headers:
+    "X-Fastly-Country-Code" = "client.geo.country_code",
+    "X-Fastly-Region-Code"  = "client.geo.region",
+    "X-Fastly-Postal-Code"  = "client.geo.postal_code",
+
+    # Ensures that we don't leak sensitive items from URLs in Referrer headers: we will
+    # send full URL if same-origin and HTTPS, origin-only (e.g. 'www.dosomething.org')
+    # when crossing domains, and nothing at all when going to HTTP.
+    "Referrer-Policy" = "\"strict-origin-when-cross-origin\""
+
+    # Enforce HTTPS for all traffic (TODO: https://www.pivotaltracker.com/story/show/174911098)
+    "Strict-Transport-Security" = "\"max-age=300; includeSubDomains\"" # 5 minutes.
```

Comment:
Since this can't easily be undone once someone receives this header, we'll want to test this with a short TTL for a while, and then increase until we reach 2 years long following [these best practices](https://hstspreload.org/#deployment-recommendations). (And we can consider submitting our domain for HSTS preload then, too!)

Reply:
Does this mean the header is only like active for 5 minutes? I'm a little confused about this

Reply:
It's sort-of like a cache header – this header, as written here, will be sent on all responses from www.dosomething.org. When a user's browser sees it on a response, it'll make a note that it's _only_ allowed to visit our site via HTTPS for the next five minutes. (So if someone tries to visit `http://` directly or omits a protocol, it'll force that to `https://`).

After five minutes, it'll revert to default behavior (defaulting to `http://`) until it sees this header on another response & once again forces HTTPS traffic for the next five minutes. (And hence why we'll want to bump this up after testing.)

Reply:
Ah got it! So if something does end up not working, it'll only be borked for five minutes?

Reply:
Correct!